movement/io/validators/datasets.py

```diff
@@ -0,0 +1,178 @@
+"""`attrs` classes for validating data structures."""
+
+from collections.abc import Iterable
+from typing import Any, Optional, Union
+
+import numpy as np
+from attrs import converters, define, field, validators
+
+from movement.logging import log_error, log_warning
+
+
+def _list_of_str(value: Union[str, Iterable[Any]]) -> list[str]:
+    """Try to coerce the value into a list of strings."""
+    if isinstance(value, str):
+        log_warning(
+            f"Invalid value ({value}). Expected a list of strings. "
+            "Converting to a list of length 1."
+        )
+        return [value]
+    elif isinstance(value, Iterable):
+        return [str(item) for item in value]
+    else:
+        raise log_error(
+            ValueError, f"Invalid value ({value}). Expected a list of strings."
+        )
+
+
+def _ensure_type_ndarray(value: Any) -> None:
+    """Raise ValueError the value is a not numpy array."""
+    if not isinstance(value, np.ndarray):
+        raise log_error(
+            ValueError, f"Expected a numpy array, but got {type(value)}."
+        )
+
+
+def _set_fps_to_none_if_invalid(fps: Optional[float]) -> Optional[float]:
+    """Set fps to None if a non-positive float is passed."""
+    if fps is not None and fps <= 0:
+        log_warning(
+            f"Invalid fps value ({fps}). Expected a positive number. "
+            "Setting fps to None."
+        )
+        return None
```

Comment:
You don't need return None. To be consisten with the other methods :)

Reply:
thanks for the feedback but I think this is desired behaviour. @niksirbi would you mind confirming?

Reply:
Yeah I think that's how `attrs` converters are supposed to work, they convert one value to another, by returning the new value. But even if it would still work without the `return None`, in this particular case I'd favour keeping it in the interest of being more explicit/readable.